File: rules/java/tests/test_mass_assignment_tn.java
// True negative: DTO with explicit field mapping (safe)
@RestController
public class UserController {

    @Autowired
    private UserRepository userRepo;

    @PostMapping("/users")
    // ok: ez-spring-mass-assignment
    public User createUser(@RequestBody CreateUserDto dto) {
        User user = new User();
        user.setName(dto.getName());
        user.setEmail(dto.getEmail());
        return userRepo.save(user);
    }
}

File: rules/php/tests/test_mass_assignment_tn.php
<?php
// True negative: validated input (safe)
class UserController extends Controller
{
    public function store(Request $request)
    {
        // ok: ez-laravel-mass-assignment
        $validated = $request->validate([
            'name' => 'required|string',
            'email' => 'required|email',
        ]);
        $user = User::create($validated);
        return response()->json($user);
    }
}
